### PR TITLE
cffile: inconsistent behavior of the attribute nameConflict with "makeunique" value LDEV-2872

### PR DESCRIPTION
**Pull request by cfmitrah**
https://luceeserver.atlassian.net/browse/LDEV-2872

### CODE REVIEW

**Comment by zspitzer** [comment]
I did some work on the error messages in this file too https://github.com/lucee/Lucee/pull/941

File: test/tickets/LDEV2872.cfc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+component extends="org.lucee.cfml.test.LuceeTestCase"{
+
+	function beforeAll() {
+		variables.uri = createURI("LDEV2872/");
+		if(!directoryExists("#uri#/filewrite/")) {
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:5]:
```suggestion
		if (!directoryExists("#uri#/filewrite/")) {
```

File: test/tickets/LDEV2872.cfc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			filewrite(file="#uri#/filewrite/fileExists.txt", data="Content that you need to write.");
+		}
+	}
+
+	function run( testResults , testBox ) {
+		describe( "test case for LDEV-2872", function() {
+			it(title = "cffile action=write with nameconflict=makeunique", body = function( currentSpec ) {
+				```
+				<cffile action="write" file="#uri#/filewrite/fileExists.txt" output="overwrite#randrange(0,100)#" nameconflict="makeunique">
+
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:16]:
```suggestion
```

File: test/tickets/LDEV2872.cfc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		});
+	}
+
+	private string function createURI(string calledName){
+		var baseURI = "/test/#listLast(getDirectoryFromPath(getCurrenttemplatepath()),"\/")#/";
+		return baseURI&""&calledName;
+	}
+
+	function afterAll() {
+		if(directoryExists(uri)) {
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:31]:
```suggestion
		if (directoryExists(uri)) {
```

File: test/tickets/LDEV2872.cfc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+			directorydelete(uri,true);
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:32]:
```suggestion
			directorydelete(uri, true);
```

File: test/tickets/LDEV2872.cfc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		}
+	}
+}
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:35]:
```suggestion
}

```

File: test/tickets/LDEV2872.cfc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+				```
+				len = arraylen(directorylist("#uri#/filewrite/"));
+				expect(len).toBeGT(1, len);
+			});
+
+		});
+	}
+
+	private string function createURI(string calledName){
+		var baseURI = "/test/#listLast(getDirectoryFromPath(getCurrenttemplatepath()),"\/")#/";
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:26]:
```suggestion
		var baseURI = "/test/#listLast(getDirectoryFromPath(getCurrenttemplatepath()), "\/")#/";
```

File: test/tickets/LDEV2872.cfc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+	function beforeAll() {
+		variables.uri = createURI("LDEV2872/");
+		if(!directoryExists("#uri#/filewrite/")) {
+			directorycreate("#uri#/filewrite/");
+			filewrite(file="#uri#/filewrite/fileExists.txt", data="Content that you need to write.");
+		}
+	}
+
+	function run( testResults , testBox ) {
```
Comment (jbampton) [inline test/tickets/LDEV2872.cfc:11]:
```suggestion
	function run(testResults, testBox) {
```

**Pull request merged**